include/zephyr/net/lwm2m.h

```diff
@@ -1205,6 +1205,25 @@ int lwm2m_engine_set_time(const char *pathstr, time_t value);
  */
 int lwm2m_set_time(const struct lwm2m_obj_path *path, time_t value);
 
+/**
+ * @brief LwM2M resource item structure
+ */
+struct lwm2m_res_item {
+	struct lwm2m_obj_path *path; /**< Pointer to LwM2M path as a struct */
+	void *value; /**< Pointer to resource value */
+	uint16_t len; /**< Length of value */
+};
+
+/**
+ * @brief Set multiple resource (instance) values
+ *
+ * @param[in] res_list LwM2M resource item list
+ * @param[in] res_list_size Length of resource list
+ *
+ * @return 0 for success or negative in case of error.
+ */
```

Comment:
```suggestion
 * @brief Set multiple resource (instance) values
 *
 * When writing strings to opaque resources, the null-terminator should not be counted in the size field.
 * When writing strings to string resources, the null-terminator should be included in the size.
 *
 * @param[in] res_list LwM2M resource item list
 * @param[in] res_list_size Length of resource list
 *
 * @return 0 for success or negative in case of error.
 */
```